functions/utils/getToken.ts

```diff
@@ -11,23 +12,46 @@ function formatTitleName(symbol: string, name: string) {
   return 'View Token on Uniswap'
 }
 
-export default async function getToken(networkName: string, tokenAddress: string | undefined, url: string) {
-  const { data } = await client.query({
+const convertTokenAddress = (networkName: string, tokenAddress: string) => {
+  if (tokenAddress === 'NATIVE') {
+    switch (networkName) {
+      case Chain.Celo:
+        return '0x471EcE3750Da237f93B8E339c536989b8978a438'
+      case Chain.Polygon:
+        return '0x0000000000000000000000000000000000001010'
+      default:
+        return undefined
+    }
+  }
+  return tokenAddress
+}
+
+export default async function getToken(networkName: string, tokenAddress: string, url: string) {
+  const origin = new URL(url).origin
+  const image = origin + '/api/image/tokens/' + networkName + '/' + tokenAddress
+  const uppercaseNetworkName = networkName.toUpperCase()
+  const convertedTokenAddress = convertTokenAddress(uppercaseNetworkName, tokenAddress)
+  const { data } = await client.query<TokenQuery>({
     query: TokenDocument,
     variables: {
-      chain: networkName,
-      address: tokenAddress,
+      chain: uppercaseNetworkName,
+      address: convertedTokenAddress,
     },
   })
   const asset = data?.token
   if (!asset) {
     return undefined
   }
+
   const title = formatTitleName(asset.symbol, asset.name)
+
   const formattedAsset = {
     title,
-    image: asset.project?.logoUrl,
+    image,
     url,
+    symbol: asset.symbol ?? 'UNK',
+    ogImage: asset.project?.logoUrl ?? '',
```

Comment:
Can this just be left as undefined, and then you can simplify the validator and update the logic in rendering to match?